### DISCUSSION TITLE
[Bitcoin Calculate Fee] How can i calculate fee on Android or iOS.

### DISCUSSION THREAD

**Discussion by weixuefeng** [trustwallet/wallet-core#3287] [Q&A] [answered]
i make brc20 transfer transaction. but how can i calculate transaction fee? or how can i calculate transaction size?

such as this transaction: https://github.com/trustwallet/wallet-core/blob/bad5dabe0b02c658e126c21a3e963b0c4b1ebae3/android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/bitcoin/TestBitcoinSigning.kt#L164

**Comment by weixuefeng** [comment]
@satoshiotomakan

**Comment by Milerius** [comment]
You need to use online rpc to calculate the fees @weixuefeng - fees calculation is not included in wallet core for any chains

**Comment by weixuefeng** [comment]
@Milerius if my transaction is a normal transaction, just like from A to B, not brc20 transaction, i can use `feeRate` and transaction plan calculate bitcoin transaction fee, but what should i do when transaction is brc20, i found related method about `FeeCalculator`, maybe do you know how can i calculate transaction visual size?

**Comment by weixuefeng** [comment, marked as answer]
- https://jlopp.github.io/bitcoin-transaction-size-calculator/
- https://bitcoinops.org/en/tools/calc-size/
